src/irmin-pack/inode_intf.ml

```diff
@@ -60,6 +80,9 @@ module type S = sig
        and type Portable.hash := hash
 
   val decode_bin_length : string -> int -> int
+  val snapshot_iter : value -> Val.Snapshot.t
+  val snapshot_add : 'a t -> Val.Snapshot.t -> key
```

Comment:
It seems that `snapshot_add` internally converts a snapshot to an inode before calling `save` on it.

Wouldn't it make more sense provide an `of_snapshot` function instead of `snapshot_add`? It would then be possible to call `add` on it.